tests/unit/test_loaders.py

```diff
@@ -87,39 +132,63 @@ def test_default_search_paths(self):
         # with .aws/models.
         home_dir_path = os.path.join('.aws', 'models')
         self.assertTrue(
-            any(p.endswith(home_dir_path) for p in loader.search_paths)
+            any(str(p).endswith(home_dir_path) for p in loader.search_paths)
         )
+        self.assertIsInstance(loader.search_paths, SearchPathList)
 
     def test_can_add_to_search_path(self):
         loader = Loader()
         loader.search_paths.append('mypath')
-        self.assertIn('mypath', loader.search_paths)
+        self.assertIn(pathlib.Path('mypath'), loader.search_paths)
+        self.assertIsInstance(loader.search_paths, SearchPathList)
 
     def test_can_initialize_with_search_paths(self):
         loader = Loader(extra_search_paths=['foo', 'bar'])
         # Note that the extra search paths are before
         # the customer/builtin data paths.
+
+        self.assertIsInstance(loader.search_paths, SearchPathList)
         self.assertEqual(
             loader.search_paths,
             [
-                'foo',
-                'bar',
+                pathlib.Path('foo'),
+                pathlib.Path('bar'),
                 loader.CUSTOMER_DATA_PATH,
                 loader.BUILTIN_DATA_PATH,
             ],
         )
 
+    def test_cant_add_wrong_type_to_search_path(self):
+
+        bad_types = (
+            None,
+            [1],
+            1,
+            1.5,
+            {'bad': 'type'},
+            b'bad_type',
+            False,
+            {'bad_type'},
+        )
```

Comment:
Having these check lists inside tests usually makes debugging failures difficult. Can we move this out to a parametrize decorator? It's not actually using any of the instance state for the test.

Reply:
yup will do